build.sbt

```diff
@@ -301,8 +301,26 @@ lazy val fs2 = project
     )
   )
 
+lazy val tasks = project
```

Comment:
This seems like quite a core feature/construct - i'd actually be tempted to add this change into the `core` project and remove the dependency on FS2 (which is only used for tests by the looks)

Reply:
Agreed. I see no harm in extending the core API with a more high level construct

Reply:
Hm. I'm not sure I agree. 

Specifically `TaskTestOps` has two gnarly implementations for creating stub consume handlers - one for `Lifecycle` and one for `fs2`. Unfortunately we didn't manage to generalise those two to a single method and in AM world we rely extensively on both implementations. 

If we were to include this into core, we'd still have to provide a `TaskTestOps` (or similar) in core and a `TaskFs2TestOps` in `bucky.fs2`

Reply:
I'm worried there are too many projects being added to bucky, and as this doesn't pull in any dependencies reckon it should be in core (or rabbitmq.. but I think core)

You could add the main constructs into the `core` and test coverage into the `rabbitmq` project, and test in some way similar to this:
https://github.com/ITV/bucky/blob/master/rabbitmq/src/test/scala/com/itv/bucky/future/GenericPublisherTest.scala

Reply:
If we include this in the core API, would it lead to clients using constructs from Task and from the lower level API arbitrarily? Another option might be to have a `bucky-extensions` sub/sibling project that contains abstractions like this. I'm keen that we get these changes merged in as soon as possible, so could this decision be deferred? And possibly logged as a GH issue?